cljfmt/src/cljfmt/core.cljc

```diff
@@ -71,21 +77,114 @@
        (not (namespaced-map? (z/up* zloc)))
        (element? (z/right* zloc))))
 
+(def ^:private default-align-bindings-args
+  (read-resource "cljfmt/align_bindings/clojure.clj"))
+
+(defn- ks->max-length [ks]
+  (if (empty? ks)
+    0
+    (->> ks
+         (apply max-key (comp count str))
+         str
+         count)))
+
+(defn- aligner [zloc max-length align?]
+  (cond
+    (zero? max-length)  (z/up zloc)
+    (z/rightmost? zloc) (z/up zloc)
+    align?              (let [clean-zloc (-> zloc
+                                             z/right*
+                                             (z/replace (n/whitespace-node " "))
+                                             z/left)
+                              to-add     (->> clean-zloc
+                                              z/sexpr
+                                              str
+                                              count
+                                              (- max-length))
+                              new-zloc   (z/insert-space-right clean-zloc to-add)]
+                          (aligner (z/right new-zloc) max-length false))
+    :else               (aligner (z/right zloc) max-length true)))
+
+(defn- align-binding [zloc]
+  (let [se         (z/sexpr zloc)
+        ks         (take-nth 2 se)
+        max-length (ks->max-length ks)
+        bindings   (z/down zloc)]
+    (if bindings
+      (aligner bindings max-length true)
+      zloc)))
+
+(defn- align-map [zloc]
+  (let [se         (z/sexpr zloc)
+        ks         (keys se)
+        max-length (ks->max-length ks)
+        kvs        (z/down zloc)]
+    (if kvs
+      (aligner kvs max-length true)
+      zloc)))
+
+(defn- sibling-distance [left right]
+  (if (= left right)
+    0
+    (if (z/rightmost? left)
+      nil
+      (when-let [d (sibling-distance (z/right left) right)]
+        (+ 1 d)))))
+
+(defn- binding? [zloc align-bindings-args]
+  (and (z/vector? zloc)
+       (-> zloc z/sexpr count even?)
+       (let [sexpr-type (-> zloc
+                            z/leftmost
+                            z/value)
+             zloc-pos   (-> zloc
+                            z/leftmost
+                            (sibling-distance zloc)
+                            dec)
+             align-args (align-bindings-args sexpr-type)]
+         (and align-args
+              (align-args zloc-pos)))))
+
+(defn- sexpr-able? [zloc]
```

Comment:
@weavejester  not sure how do make it nicer. Any idea?

Without it the "abnormal" tests for "cljs map values" fails. I never used this syntax and it's hard to google so I'm not sure what do to exactly. z/sexpr-able? returns true but z/sexpr throws an error for not supplying value to the reader thing.

I found it by running cljfmt on itself. I will run on more projects later.